### ISSUE TITLE
None Selected

### ISSUE THREAD

**Issue by 444B** [444B/streamlit-analytics2#13]
Original issue by hansipie on 2024-02-02 18:22:02+00:00

Hello, 

Defining a selectbox like this :
`select = st.selectbox("title", ("A", "B"), index=None)`
... I get a KeyError exception at main.py:162
This is caused by the default "None" value. It is changed into an empty space by "replace_empty(...)", but
`if selected != st.session_state.state_dict.get(label, None):` goes into the "true" statement, and causes an exception.

The fix at main.py:161:
`if selected != st.session_state.state_dict.get(label, " "):`

I make a PR or let you fix ...

**Comment by 444B** [comment]
@hansipie is this still relevant?

**Comment by hansipie** [comment]
Hello.
First of all sorry for the delay ...
Yes it does... this is a sample to reproduce:
```python
import streamlit as st
import streamlit_analytics2 as streamlit_analytics

with streamlit_analytics.track():
    options = st.selectbox("options", ("option 1", "option 2"), index=None)
    st.info(options)
``` 
The fix I proposed was a first idea and needs to be explored further.

PS: Thx for this fork :)
